### ISSUE TITLE
Prevent cluster getting stuck in "upgrading" state after upgrading the rke2 version for an imported rke2 cluster

### ISSUE THREAD

**Issue by erSitzt** [rancher/rancher#38191]
**Rancher Server Setup**
- Rancher version: 2.6.6
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE2

Rancher 2.6.6 Local Cluster ( v1.21.7+rke2r2 )

I ran into this issue when upgrading my Cluster from `v1.21.7+rke2r2`  to `v1.22.10+rke2r2`
There is a similar older issue here #34067 for reference.

Most nodes did work it seems, but now its stuck with two nodes left
<img width="942" alt="image" src="https://user-images.githubusercontent.com/1625376/177279618-b5e98b51-7015-4c9f-b433-c81ff17e427f.png">

With this failed job in the cluster
<img width="1453" alt="image" src="https://user-images.githubusercontent.com/1625376/177280080-12e2acb3-9edb-47f4-87e4-035551d326f9.png">
This is the failed jobs info ( `kubectl describe`)
<details>
  <summary>Click to expand!</summary>
  
  ```
❯ kubectl describe job apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-94b29 -n cattle-system
Name:                     apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-94b29
Namespace:                cattle-system
Selector:                 controller-uid=7f1c561d-09c0-4bd5-ae29-959bd2faff63
Labels:                   objectset.rio.cattle.io/hash=6641dc529612dc10f1658bf00333df03480e1367
                          plan.upgrade.cattle.io/rke2-worker-plan=75375c2bb1dd80940ce0f2f48102d8f4aef7c8b3fb6febf3cd9f82a1
                          upgrade.cattle.io/controller=system-upgrade-controller
                          upgrade.cattle.io/node=rke2-downstream1-agent-2
                          upgrade.cattle.io/plan=rke2-worker-plan
                          upgrade.cattle.io/version=v1.22.10-rke2r2
Annotations:              batch.kubernetes.io/job-tracking:
                          objectset.rio.cattle.io/applied:
                            H4sIAAAAAAAA/+xY227bOBN+lf/nteRKspPYBvbCG7tbo41j1GkXRREEFDmyuaFILTlyYhh+9wUp2ZEbO4cuFuhFUKARKc5wDt83mvGa5ICUU6SkvyZUKY0UhVbWLXX6FzC0gC0jdI...
                          objectset.rio.cattle.io/id: system-upgrade-controller
                          objectset.rio.cattle.io/owner-gvk: upgrade.cattle.io/v1, Kind=Plan
                          objectset.rio.cattle.io/owner-name: rke2-worker-plan
                          objectset.rio.cattle.io/owner-namespace: cattle-system
                          upgrade.cattle.io/ttl-seconds-after-finished: 900
Parallelism:              1
Completions:              1
Completion Mode:          NonIndexed
Start Time:               Tue, 05 Jul 2022 09:45:47 +0200
Active Deadline Seconds:  900s
Pods Statuses:            0 Active / 0 Succeeded / 1 Failed
Pod Template:
  Labels:           controller-uid=7f1c561d-09c0-4bd5-ae29-959bd2faff63
                    job-name=apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-94b29
                    plan.upgrade.cattle.io/rke2-worker-plan=75375c2bb1dd80940ce0f2f48102d8f4aef7c8b3fb6febf3cd9f82a1
                    upgrade.cattle.io/controller=system-upgrade-controller
                    upgrade.cattle.io/node=rke2-downstream1-agent-2
                    upgrade.cattle.io/plan=rke2-worker-plan
                    upgrade.cattle.io/version=v1.22.10-rke2r2
  Service Account:  system-upgrade-controller
  Init Containers:
   prepare:
    Image:      rancher/rke2-upgrade:v1.22.10-rke2r2
    Port:       <none>
    Host Port:  <none>
    Args:
      prepare
      rke2-master-plan
    Environment:
      SYSTEM_UPGRADE_NODE_NAME:             (v1:spec.nodeName)
      SYSTEM_UPGRADE_POD_NAME:              (v1:metadata.name)
      SYSTEM_UPGRADE_POD_UID:               (v1:metadata.uid)
      SYSTEM_UPGRADE_PLAN_NAME:            rke2-worker-plan
      SYSTEM_UPGRADE_PLAN_LATEST_HASH:     75375c2bb1dd80940ce0f2f48102d8f4aef7c8b3fb6febf3cd9f82a1
      SYSTEM_UPGRADE_PLAN_LATEST_VERSION:  v1.22.10-rke2r2
    Mounts:
      /host from host-root (rw)
      /run/system-upgrade/pod from pod-info (ro)
   drain:
    Image:      rancher/kubectl:v1.20.2
    Port:       <none>
    Host Port:  <none>
    Args:
      drain
      rke2-downstream1-agent-2
      --pod-selector
      !upgrade.cattle.io/controller
      --ignore-daemonsets
      --delete-local-data
      --force
    Environment:
      SYSTEM_UPGRADE_NODE_NAME:             (v1:spec.nodeName)
      SYSTEM_UPGRADE_POD_NAME:              (v1:metadata.name)
      SYSTEM_UPGRADE_POD_UID:               (v1:metadata.uid)
      SYSTEM_UPGRADE_PLAN_NAME:            rke2-worker-plan
      SYSTEM_UPGRADE_PLAN_LATEST_HASH:     75375c2bb1dd80940ce0f2f48102d8f4aef7c8b3fb6febf3cd9f82a1
      SYSTEM_UPGRADE_PLAN_LATEST_VERSION:  v1.22.10-rke2r2
    Mounts:
      /host from host-root (rw)
      /run/system-upgrade/pod from pod-info (ro)
  Containers:
   upgrade:
    Image:      rancher/rke2-upgrade:v1.22.10-rke2r2
    Port:       <none>
    Host Port:  <none>
    Environment:
      SYSTEM_UPGRADE_NODE_NAME:             (v1:spec.nodeName)
      SYSTEM_UPGRADE_POD_NAME:              (v1:metadata.name)
      SYSTEM_UPGRADE_POD_UID:               (v1:metadata.uid)
      SYSTEM_UPGRADE_PLAN_NAME:            rke2-worker-plan
      SYSTEM_UPGRADE_PLAN_LATEST_HASH:     75375c2bb1dd80940ce0f2f48102d8f4aef7c8b3fb6febf3cd9f82a1
      SYSTEM_UPGRADE_PLAN_LATEST_VERSION:  v1.22.10-rke2r2
    Mounts:
      /host from host-root (rw)
      /run/system-upgrade/pod from pod-info (ro)
  Volumes:
   host-root:
    Type:          HostPath (bare host directory volume)
    Path:          /
    HostPathType:  Directory
   pod-info:
    Type:  DownwardAPI (a volume populated by information about the pod)
    Items:
      metadata.labels -> labels
      metadata.annotations -> annotations
Events:
  Type     Reason            Age                  From            Message
  ----     ------            ----                 ----            -------
  Normal   SuccessfulCreate  22m                  job-controller  Created pod: apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-6xlpz
  Normal   SuccessfulDelete  7m6s                 job-controller  Deleted pod: apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-6xlpz
  Warning  DeadlineExceeded  7m6s (x2 over 7m6s)  job-controller  Job was active longer than specified deadline
  ```
</details>

And in the cluster view
<img width="569" alt="image" src="https://user-images.githubusercontent.com/1625376/177279758-78f63623-52e6-4812-a383-3f32f71c9a79.png">

**Comment by Oats87** [comment]
@erSitzt When you look at the logs for that job, do you see anything that sticks out?

That job is what we use with the system-upgrade-controller with https://github.com/rancher/rke2-upgrade to perform the actual upgrade of RKE2.

**Comment by erSitzt** [comment]
Do you mean the logs of the pod that was started by the job ?

The pod was already deleted:
```
Events:
  Type     Reason            Age    From            Message
  ----     ------            ----   ----            -------
  Normal   SuccessfulCreate  18m    job-controller  Created pod: apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-kxt6f
  Normal   SuccessfulDelete  3m23s  job-controller  Deleted pod: apply-rke2-worker-plan-on-rke2-downstream1-agent-2-with-7-kxt6f
  Warning  DeadlineExceeded  3m23s  job-controller  Job was active longer than specified deadline
```

**Comment by erSitzt** [comment]
If there is anything useful i can provide, i can leave the cluster in this state, otherwise i could try to update to the next version and get the logs while the update is running...

<img width="877" alt="image" src="https://user-images.githubusercontent.com/1625376/177578324-095c5ee2-7bb9-44a8-a005-06492cb106c1.png">

**Comment by erSitzt** [comment]
not stale

**Comment by Martin-Weiss** [comment]
@erSitzt - are you still stuck in this issue? I believe the root cause seems to be clear - the PDB has blocked the drain and the drain timed out.. or do I miss something? So what type of help do you need, now?

**Comment by erSitzt** [comment]
Let me rephrase the issue

**Comment by Martin-Weiss** [comment]
Ok - so changing the default job timeouts should be one thing that should be done..  that gives more time to debug and troubleshoot in case of PDBs blocking or other issues are there..

The other thing to check "are all applications implemented in a fault tolerant way with nothing having replica=1" is something we could add to documentation I assume, even though I assume this might be considered as kubernetes best practices..

**Comment by erSitzt** [comment]
not stale...
